dd-sdk-android/src/main/kotlin/com/datadog/android/webview/internal/rum/WebViewRumEventConsumer.kt

```diff
@@ -29,75 +25,34 @@ internal class WebViewRumEventConsumer(
-            if (mappedEvent != null) {
-                dataWriter.write(mappedEvent)
-            }
-        } else {
-            dataWriter.write(event)
-        }
+        val mappedEvent = map(event, rumContext)
+        dataWriter.write(mappedEvent)
     }
 
     private fun map(
         event: JsonObject,
-        eventType: String,
-        rumContext: RumContext
-    ): JsonObject? {
+        rumContext: RumContext?
+    ): JsonObject {
         return try {
-            when (eventType) {
-                VIEW_EVENT_TYPE -> {
-                    val parsedViewEvent = ViewEvent.fromJson(event.toString())
-                    webViewRumEventMapper.mapViewEvent(
-                        parsedViewEvent,
-                        rumContext,
-                        getOffset(parsedViewEvent.view.id)
-                    ).toJson().asJsonObject
-                }
-                ACTION_EVENT_TYPE -> {
-                    val parsedActionEvent = ActionEvent.fromJson(event.toString())
-                    webViewRumEventMapper.mapActionEvent(
-                        parsedActionEvent,
-                        rumContext,
-                        getOffset(parsedActionEvent.view.id)
-                    ).toJson().asJsonObject
-                }
-                RESOURCE_EVENT_TYPE -> {
-                    val parsedResourceEvent = ResourceEvent.fromJson(event.toString())
-                    webViewRumEventMapper.mapResourceEvent(
-                        parsedResourceEvent,
-                        rumContext,
-                        getOffset(parsedResourceEvent.view.id)
-                    ).toJson().asJsonObject
-                }
-                ERROR_EVENT_TYPE -> {
-                    val parsedErrorEvent = ErrorEvent.fromJson(event.toString())
-                    webViewRumEventMapper.mapErrorEvent(
-                        parsedErrorEvent,
-                        rumContext,
-                        getOffset(parsedErrorEvent.view.id)
-                    ).toJson().asJsonObject
-                }
-                LONG_TASK_EVENT_TYPE -> {
-                    val parsedLongTaskEvent = LongTaskEvent.fromJson(event.toString())
-                    webViewRumEventMapper.mapLongTaskEvent(
-                        parsedLongTaskEvent,
-                        rumContext,
-                        getOffset(parsedLongTaskEvent.view.id)
-                    ).toJson().asJsonObject
-                }
-                else -> {
-                    sdkLogger.e(WRONG_EVENT_TYPE_ERROR_MESSAGE.format(US, eventType))
-                    null
-                }
-            }
-        } catch (e: JsonParseException) {
+            val timeOffset = event.get(VIEW_KEY_NAME)?.asJsonObject?.get(VIEW_ID_KEY_NAME)
+                ?.asString?.let { getOffset(it) } ?: 0L
+            webViewRumEventMapper.mapEvent(event, rumContext, timeOffset)
+        } catch (e: ClassCastException) {
```

Comment:
the `JsonPrimitve#asLong` methods states that can throw this exception. For now I could not see where and how can we reproduce this. I still think we need to keep it here just in case.